spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/config/MongoDbOutboundGatewayParserTests.java

```diff
@@ -109,6 +110,20 @@ public void fullConfigWithMongoTemplate() {
 		assertEquals("foo", TestUtils.getPropertyValue(gateway, "collectionNameExpression.literalValue"));
 	}
 
+	@Test
+	public void fullConfigWithCollectionCallback() {
+		MongoDbOutboundGateway gateway = TestUtils.getPropertyValue(
+				context.getBean("fullConfigWithCollectionCallback"), "handler", MongoDbOutboundGateway.class);
+
+		assertEquals(context.getBean("mongoDbTemplate"), TestUtils.getPropertyValue(gateway, "mongoTemplate"));
```

Comment:
`assertSame()` would be better, because we really compare variables for the same object.